pandas/tseries/tests/test_resample.py

```diff
@@ -620,6 +627,71 @@ def test_resample_interpolate(self):
             df.resample('1T').asfreq().interpolate(),
             df.resample('1T').interpolate())
 
+    def test_resample_empty_series(self):
+        # GH12771 & GH12868
+
+        s = self.create_series()[:0]
+
+        for freq in ['M', 'D', 'H']:
+            # need to test for ohlc from GH13083
+            methods = [method for method in resample_methods
+                       if method != 'ohlc']
+            for method in methods:
+                expected_index = s.index._shallow_copy(freq=freq)
+
+                result = getattr(s.resample(freq), method)()
+                expected = s
+                assert_index_equal(result.index, expected_index)
+                # freq equality not yet checked in assert_index_equal
+                self.assertEqual(result.index.freq, expected_index.freq)
+                if (method == 'size' and
+                   isinstance(result.index, PeriodIndex) and
+                   freq in ['M', 'D']):
+                    # GH12871 - TODO: name should propagate, but currently
+                    # doesn't on lower / same frequency with PeriodIndex
+                    assert_series_equal(result, expected, check_dtype=False,
+                                        check_names=False)
+                    # this assert will break when fixed
+                    self.assertTrue(result.name is None)
+                else:
+                    assert_series_equal(result, expected, check_dtype=False)
+
+    def test_resample_empty_dataframe(self):
+        # GH13212
+        index = self.create_series().index[:0]
+        f = DataFrame(index=index)
+
+        for freq in ['M', 'D', 'H']:
+            # count retains dimensions too
+            methods = downsample_methods + ['count']
+            for method in methods:
+                expected_index = f.index._shallow_copy(freq=freq)
+                result = getattr(f.resample(freq), method)()
+                expected = f
+                assert_index_equal(result.index, expected_index)
+                # freq equality not yet checked in assert_index_equal
```

Comment:
Issued #13242 for tracking purpose. I already have ``.freq`` testing branch and it works except this empty resampe issue.